cpp/src/plasma/plasma/plasma.pyx

```diff
@@ -0,0 +1,323 @@
+    self.data = CUniqueID.from_binary(object_id)
+
+  def __richcmp__(ObjectID self, ObjectID object_id, operation):
+    assert operation == 2, "only equality implemented so far"
+    return self.data == object_id.data
+
+  def __repr__(self):
+    return "ObjectID(" + self.data.hex().decode() + ")"
+
+  def __reduce__(self):
+    return (make_object_id, (self.data.binary(),))
+
+cdef class PlasmaBuffer(Buffer):
+  """This is the type of objects returned by calls to get with a PlasmaClient.
+
+  We define our own class instead of directly returning a buffer object so that
+  we can add a custom destructor which notifies Plasma that the object is no
+  longer being used, so the memory in the Plasma store backing the object can
+  potentially be freed.
+
+  Attributes:
+    object_id (ObjectID): The ID of the object in the buffer.
+    client (PlasmaClient): The PlasmaClient that we use to communicate
+      with the store and manager.
+  """
+
+  cdef:
+    ObjectID object_id
+    PlasmaClient client
+
+  def __cinit__(self, ObjectID object_id, PlasmaClient client):
+    """Initialize a PlasmaBuffer."""
+    self.object_id = object_id
+    self.client = client
+
+  def __dealloc__(self):
+    """Notify Plasma that the object is no longer needed.
+
+    If the plasma client has been shut down, then don't do anything.
+    """
+    self.client.release(self.object_id)
+
+cdef class PlasmaClient:
+  """The PlasmaClient is used to interface with a plasma store and manager.
+
+  The PlasmaClient can ask the PlasmaStore to allocate a new buffer, seal a
+  buffer, and get a buffer. Buffers are referred to by object IDs, which are
+  strings.
+  """
+
+  cdef:
+    shared_ptr[CPlasmaClient] client
+    int notification_fd
+
+  def __cinit__(self):
+    self.client.reset(new CPlasmaClient())
+    self.notification_fd = -1
+
+  cdef _get_object_buffers(self, object_ids, int64_t timeout_ms, c_vector[CObjectBuffer]* result):
+    cdef c_vector[CUniqueID] ids
+    cdef ObjectID object_id
+    for object_id in object_ids:
+      ids.push_back(object_id.data)
+    result[0].resize(ids.size())
+    check_status(self.client.get().Get(ids.data(), ids.size(), timeout_ms, result[0].data()))
+
+  cdef _make_plasma_buffer(self, ObjectID object_id, uint8_t* data, int64_t size):
+    cdef shared_ptr[MutableBuffer] buffer
+    buffer.reset(new MutableBuffer(data, size))
+    result = PlasmaBuffer(object_id, self)
+    result.init(<shared_ptr[CBuffer]>(buffer))
+    return result
+
+  def connect(self, store_socket_name, manager_socket_name, release_delay):
+    """Conect the PlasmaClient to a plasma store and optionally a manager.
+
+    Args:
+      store_socket_name (str): Name of the socket the plasma store is listening
+        at.
+      manager_socket_name (str): Name of the socket the plasma manager is
+        listening at.
+      release_delay (int): The maximum number of objects that the client will
+        keep and delay releasing (for caching reasons).
```

Comment:
Can you use numpydoc style here and in the other docstrings? https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt